Home Work N7

DmytriiTsiupko opened the pull request:
1. Use the file as a database to store phonebook entries from previous tasks.

Your phonebook, previously contained in a dict, should be stored as text in JSON format. When closing the program and opening it again, all previous data should be available.

2. Write a decorator that will write to a file the name of the function it decorates and write the time of its call.

3. In the previously written custom Exception, add a record of the error and the time of its occurrence to the file.

DmytriiTsiupko committed:
Home Work N7

1. Use the file as a database to store phonebook entries from previous tasks.

Your phonebook, previously contained in a dict, should be stored as text in JSON format.
When closing the program and opening it again, all previous data should be available.

2. Write a decorator that will write to a file the name of the function it decorates and write the time of its call.

3. In the previously written custom Exception, add a record of the error and the time of its occurrence to the file.

acman commented on 07. Files/01_ContactBook/01_ContactBook.py:
```diff
@@ -0,0 +1,108 @@
+import os
+
+
+with open("contacts.json", "a+") as file:
```
nit: it's a good practice to set encoding

acman commented on 07. Files/01_ContactBook/01_ContactBook.py:
```diff
@@ -0,0 +1,108 @@
+    with open("contacts.json", "r+") as file_data:
+        our_contacts = json.loads(file_data.read())
+
+    with open("contacts.json", "w") as file_data:
```
nit: your file is already open using `r+` so you can move to the start of the file using `seek` and write to it and not open it again.

acman commented on 07. Files/02_LogDecorator/02_LogDecorator.py:
```diff
@@ -0,0 +1,21 @@
+def get_log(func):
+    def wraps(*args, **kwargs):
+        with open("logs.txt", "a") as file:
+            print(f"{func} was called at {datetime.datetime.now()}", file=file)
```
nit:` func.__name__` could be used here if you want show only name of function

acman reviewed (approved):
LGTM! Would be great if you play with `seek`, `tell`, `truncate` to understand navigation inside file

DmytriiTsiupko committed:
Homework N7 (Files)

Changed code:
- added encoding to creating a file (01_ContactBook);
- use "seek(0)" instead of a one more "open()" function (01_ContactBook);
- use "func.__name__" to show name of function by loging
